src/algorithms/pso.cpp

```diff
@@ -198,25 +208,31 @@ population pso::evolve(population pop) const
     }
 
     // Initialize the Swarm's topology
-    switch (m_neighb_type) {
-        case 1:
-            initialize_topology__gbest(pop, best_neighb, best_fit, neighb);
-            break;
-        case 3:
-            initialize_topology__von(neighb);
-            break;
-        case 4:
-            initialize_topology__adaptive_random(neighb);
-            // need to track improvements in best found fitness, to know when to rewire
-            best_fit = pop.get_f()[pop.best_idx()];
-            break;
-        case 2:
-        default:
-            initialize_topology__lbest(neighb);
+    if (m_memory && m_memory_data) {
+        neighb = m_memory_data->m_neighb;
+        best_fit = m_memory_data->m_best_fit;
+        best_neighb = m_memory_data->m_best_neighb;
+    } else {
+        switch (m_neighb_type) {
+            case 1:
+                initialize_topology__gbest(pop, best_neighb, best_fit, neighb);
+                break;
+            case 3:
+                initialize_topology__von(neighb);
+                break;
+            case 4:
+                initialize_topology__adaptive_random(neighb);
+                // need to track improvements in best found fitness, to know when to rewire
+                best_fit = pop.get_f()[pop.best_idx()];
+                break;
+            case 2:
+            default:
+                initialize_topology__lbest(neighb);
+        }
     }
     // auxiliary variables specific to the Fully Informed Particle Swarm variant
     double acceleration_coefficient = m_eta1 + m_eta2;
-    double sum_forces;
```

Comment:
Unrelated. It was just that the variable was not initialized.
It is not a bug, since it is initialized later in the code, but I find it reasonable to avoid uninitialized stuff.